src/wasm-lib/kcl/src/std/appearance.rs

```diff
@@ -0,0 +1,260 @@
+//! Standard library appearance.
+
+use anyhow::Result;
+use derive_docs::stdlib;
+use kcmc::{each_cmd as mcmd, ModelingCmd};
+use kittycad_modeling_cmds::{self as kcmc, shared::Color};
+use regex::Regex;
+use rgba_simple::Hex;
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+use validator::Validate;
+
+use crate::{
+    errors::{KclError, KclErrorDetails},
+    execution::{ExecState, KclValue, Solid, SolidSet},
+    std::Args,
+};
+
+/// Data for appearance.
+#[derive(Debug, Clone, Deserialize, Serialize, PartialEq, ts_rs::TS, JsonSchema, Validate)]
+#[ts(export)]
+#[serde(rename_all = "camelCase")]
+pub struct AppearanceData {
+    /// Color of the new material, a hex string like "#ff0000".
+    #[schemars(regex(pattern = "#[0-9a-fA-F]{6}"))]
+    pub color: String,
+    /// Metalness of the new material, a percentage like 95.7.
+    #[validate(range(min = 0.0, max = 100.0))]
+    pub metalness: Option<f64>,
+    /// Roughness of the new material, a percentage like 95.7.
+    #[validate(range(min = 0.0, max = 100.0))]
+    pub roughness: Option<f64>,
+    // TODO(jess): we can also ambient occlusion here I just don't know what it is.
+}
+
+/// Set the appearance of a solid. This only works on solids, not sketches or individual paths.
+pub async fn appearance(_exec_state: &mut ExecState, args: Args) -> Result<KclValue, KclError> {
+    let (data, solid_set): (AppearanceData, SolidSet) = args.get_data_and_solid_set()?;
+
+    // Validate the data.
+    data.validate().map_err(|err| {
+        KclError::Semantic(KclErrorDetails {
+            message: format!("Invalid appearance data: {}", err),
+            source_ranges: vec![args.source_range],
+        })
+    })?;
+
+    // Make sure the color if set is valid.
+    let re = Regex::new(r"^#[0-9a-fA-F]{6}$").unwrap();
+    if !re.is_match(&data.color) {
+        return Err(KclError::Semantic(KclErrorDetails {
+            message: format!("Invalid hex color (`{}`), try something like `#fff000`", data.color),
+            source_ranges: vec![args.source_range],
+        }));
+    }
+
+    let result = inner_appearance(data, solid_set, args).await?;
+    Ok(result.into())
+}
+
+/// Set the appearance of a solid. This only works on solids, not sketches or individual paths.
+///
+/// This will work on any solid, including extruded solids, revolved solids, and shelled solids.
+/// ```no_run
+/// /// Add color to an extruded solid.
+/// const exampleSketch = startSketchOn("XZ")
```

Comment:
We should stop using `const`. They're warnings now.

```suggestion
/// exampleSketch = startSketchOn("XZ")
```

Reply:
on it sorry i copy postad from other files